src/lib.rs

```diff
@@ -0,0 +1,418 @@
+            cargo_target_dir: None,
+            editor: None,
+            editor_args: None,
+            temporary_project_dir,
+        })
+    }
+
+    pub fn get_or_create() -> Result<Self> {
+        #[cfg(unix)]
+        let config_file_path = {
+            let config_dir = xdg::BaseDirectories::with_prefix("cargo-temp")?;
+            config_dir.place_config_file("config.toml")?
+        };
+        #[cfg(windows)]
+        let config_file_path = {
+            let config_dir = dirs::config_dir()
+                .context("Could not get config directory")?
+                .join(env!("CARGO_PKG_NAME"));
+            let _ = fs::create_dir_all(&config_dir);
+
+            config_dir.join("config.toml")
+        };
+
+        let config: Self = match fs::read(&config_file_path) {
+            Ok(file) => toml::de::from_slice(&file)?,
+            Err(_) => {
+                let config = Self::new()?;
+                fs::write(&config_file_path, toml::ser::to_string(&config)?)?;
+                println!("Config file created at: {}", config_file_path.display());
+
+                config
+            }
+        };
+
+        Ok(config)
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Dependency {
+    CrateIo(String, Option<String>),
+    Repository {
+        branch: Option<String>,
+        name: String,
+        rev: Option<String>,
+        url: String,
+    },
+}
+
+fn parse_dependency(s: &str) -> Dependency {
+    // This will change when `std::lazy` is released.
+    // See https://github.com/rust-lang/rust/issues/74465.
+    static RE: Lazy<Regex> = Lazy::new(|| {
+        Regex::new(r"^([^=]+)=(((\w+://([^:@]+(:[^@]+)?@)?[^#]+)(#branch=(.+)|#rev=(.+))?)|.+)$")
+            .unwrap()
+    });
+
+    if let Some(caps) = RE.captures(s) {
+        if let Some(url) = caps.get(4) {
+            Dependency::Repository {
+                name: caps[1].to_string(),
+                url: url.as_str().to_string(),
+                branch: caps.get(8).map(|x| x.as_str().to_string()),
+                rev: caps.get(9).map(|x| x.as_str().to_string()),
+            }
+        } else {
+            Dependency::CrateIo(caps[1].to_string(), Some(caps[2].to_string()))
+        }
+    } else {
+        Dependency::CrateIo(s.to_string(), None)
+    }
+}
+
+pub fn get_shell() -> String {
+    #[cfg(unix)]
+    {
+        env::var("SHELL").unwrap_or_else(|_| "/bin/sh".to_string())
+    }
+
+    #[cfg(windows)]
+    {
+        env::var("COMSPEC").unwrap_or_else(|_| "cmd".to_string())
+    }
+}
+
+pub fn run(cli: Cli, config: &Config) -> Result<()> {
+    let tmp_dir = generate_tmp_project(
+        cli.worktree_branch.clone(),
+        cli.project_name,
+        cli.lib,
+        config.temporary_project_dir.clone(),
+    )?;
+
+    add_dependencies_to_toml(&tmp_dir, cli.dependencies)?;
+
+    let delete_file = generate_delete_file(&tmp_dir)?;
+
+    start_shell(config, &tmp_dir)?;
+
+    exit_shell(delete_file, tmp_dir, cli.worktree_branch)?;
```

Comment:
It doesn't "exit the shell". This function is more like a cleanup / tear down function.